docs/source/upgrade_to_newest_version.md

```diff
@@ -0,0 +1,112 @@
+
+## Chaincode shim changes (Go chaincode only)
+
+The recommended approach is to vendor the shim in your v1.4 Go chaincode before making upgrades to the peers and channels. If you do this, you do not need to make any additional changes to your chaincode.
+
+If you did not vendor the shim in your v1.4 chaincode, the old v1.4 chaincode images will still technically work after upgrade, but you are in a risky state. If the chaincode image gets deleted from your environment for whatever reason, the next invoke on v2.0 peer will try to rebuild the chaincode image and you'll get an error that the shim cannot be found.
+
+At this point, you have two options:
+
+1. If the entire channel is ready to upgrade chaincode, you can upgrade the chaincode on all peers and on the channel (using either the old or new lifecycle depending on the `Application` capability level you have enabled). The best practice at this point would be to vendor the new Go chaincode shim using modules.
+
+2. If the entire channel is not yet ready to upgrade the chaincode, you can use peer environment variables to specify the v1.4 chaincode environment `ccenv` be used to rebuild the chaincode images. This v1.4 `ccenv` should still work with a v2.0 peer.
+
+## Chaincode logger (Go chaincode only)
+
+Support for user chaincodes to utilize the chaincode shim's logger via `NewLogger()` has been deprecated. Chaincodes that used the shim's `NewLogger()` must now shift to their own preferred logging mechanism.
+
+For more information, check out [Logging control](./logging-control.html#chaincode).
+
+## Peer databases upgrade
+
+The databases of all peers (which include not just the state database but the history database and other internal databases for the peer) must be rebuilt using the v2.0 data format as part of the upgrade to v2.0. To trigger the rebuild, the databases must be dropped before the peer is started.
+
+For information about how to upgrade peers, check out our documentation on [upgrading components](./upgrading_your_components.html). During the process for [upgrading your peers](./upgrading_your_components.html#upgrade-the-peers), you will need to pass a `peer node upgrade-dbs` command to drop the databases of the peer.
+
+Follow the commands to upgrade a command up until the `docker run` command you see to launch the new peer container (you can skip the step where you set an `IMAGE_TAG`, since the `upgrade dbs` command is for the v2.0 release of Fabric only). Instead of that command, run this one instead:
+
+```
+docker run -d --rm -v /opt/backup/$CORE_PEER_CONTAINER/:/var/hyperledger/production/ --name $CORE_PEER_CONTAINER hyperledger/fabric-peer:2.0 peer node upgrade-dbs
+```
+
+This will drop the databases of the peer and launch a new container with the `2.0` image tag.
+
```

Comment:
The container that runs "peer node upgrade-dbs" will exit quickly.
The rebuilding step mentioned in the next line happens when you docker run the peer for the first time on the new images, NOT when you do the peer node upgrade-dbs command above.
I'd suggest put both docker run statements here for clarify... the docker run that does "peer node upgrade-dbs", and the docker run that actually starts the peer.